spacy/cli/package.py

```diff
@@ -252,7 +252,7 @@ def get_third_party_dependencies(
                     raise regerr from None
             module_name = func_info.get("module")  # type: ignore[attr-defined]
             if module_name:  # the code is part of a module, not a --code file
-                modules.add(func_info["module"].split(".")[0])  # type: ignore[index]
+                modules.add(func_info["module"].split(".")[0])  # type: ignore
```

Comment:
Is there a specific `ignore` directive we can use here? Why does `[index]` not apply anymore?

Reply:
It's throwing an `[union-index]` error because of the newly added `Union[str,int,None]`, I've adjusted the directive. 😄

Reply:
Thanks for checking!

Reply:
@thomashacker Wait, the code says `[union-attr]`, you mention `[union-index]` in your comment. Just making sure that `[union-attr]` is the correct one - is it?

Reply:
Oh sorry! I mixed them up somehow. I meant `[union-attr]` 😄